### ISSUE TITLE
Manual use of compiled templates

### ISSUE THREAD

**Issue by mikealexander** [just-boris/handlebars-hot-loader#1]
If I do something like:

```
var template = require('someTemplate.hbs');
var model = { text: 'Hello' };
var renderedTemplate = template(model);

```

I get a console error
`Uncaught TypeError: Cannot read property 'hash' of undefined`

![image](https://cloud.githubusercontent.com/assets/4212225/13984172/7789993c-f0ec-11e5-9651-a66e3c68d061.png)

Is this use case supported?  Should I be passing an options hash to the compiled template function?

**Comment by just-boris** [comment]
Hello! 

You provided interesting use-case. Initially, that loader developed to use with Marionette and contains some code that knows about it.

You can't make plain templates hot-reloadable because you need to know about component that uses your template. How are you attaching `renderedTemplate` to DOM?

**Comment by mikealexander** [comment]
Hi! Thanks for replying.

For most cases in my app I am using the templates attached to Marionette views.  There is just one case where I wanted to render the template manually and attach it to the DOM, which is for error messages.  My function for displaying error messages looks like this:

```
var MessageContainer = function(selector, fadeDelay, fadeTime){
        this.el = selector;
        this.fadeDelay = fadeDelay;
        this.fadeTime = fadeTime;
    };

MessageContainer.prototype.displayErrorMessage = function(message){
        return $(errorMessageTemplate({ text: message })).delay(this.fadeDelay).hide('fade', this.fadeTime).appendTo(this.el);
    };
```

So, the rendered template gets appended on to a DOM element from a jQuery selector.

Now that I think about it, I should probably just create a Marionette view for error messages instead of working with templates manually. I could do something like:

```
MessageContainer.prototype.displayErrorMessage = function(message){
        var view = new Marionette.ItemView({template: errorMessageTemplate, model: new Backbone.Model({text: message})});
        view.render();
        return view.$el.delay(this.fadeDelay).hide('fade', this.fadeTime).appendTo(this.el);
    };
```

**Comment by just-boris** [comment]
Well, the best thing that I can do here is just pass this out. Manually used templates will be not hot-reloadable, but it will work at least.

I am going to publish fix for it, if this behavior is okay for you.

**Comment by mikealexander** [comment]
Yes, that behaviour is great for me.  I am not looking for manually used templates to be hot-reloadable, but just to work.

Thank-you!

**Comment by just-boris** [comment]
Fixed in https://github.com/just-boris/handlebars-hot-loader/commit/9f620049645e790fee11d3589ed071ea784d9e03 and released as new version `0.0.2`
